sdks/python/apache_beam/transforms/join/keyedpcollectiontuple.py

```diff
@@ -0,0 +1,162 @@
+
+from typing import Any, Generic, List, TypeVar
+import apache_beam as beam
+from apache_beam.typehints import typehints
+from apache_beam.coders.coders import Coder
+
+K = TypeVar("K")
+V = TypeVar("V")
+InputT = TypeVar("InputT")
+OutputT = TypeVar("OutputT")
+
+class KeyedPCollectionTuple(Generic[K]):
```

Comment:
I think it should be possible to add the Join library to Python without porting `KeyedPCollectionTuple`. In Java this is necessary for CoGroupByKey, but in Python we represent tagged/keyed PCollections as `Dict[str, PCollection]`, you can see how this works with CoGroupByKey in that transform's API docs: https://beam.apache.org/releases/pydoc/2.28.0/apache_beam.transforms.util.html#apache_beam.transforms.util.CoGroupByKey